### DISCUSSION TITLE
How to plot from vtk files using pyvista?

### DISCUSSION THREAD

**Discussion by mellow753** [pyvista/pyvista#2295] [Q&A] [answered]
Hi, I have a series of .vtk files (velocity and pressure field) from OpenFOAM, and now I'd like to visualize these files using pyvista. The following is my code:

################################
import matplotlib.pyplot as plt
import pyvista as pv

U = pv.read('U_zNormal_0.1.vtk')

mesh = pv.PolyData(U)
print(f"All arrays: {mesh.array_names}")

plotter = pv.Plotter()
plotter.add_mesh(mesh, color=True)
plotter.link_views()
plotter.show(screenshot='U.png')
###############################

I can obtain the flow field but with solid grey color, shown here:
![1](https://user-images.githubusercontent.com/57876661/156718562-c65d5d96-572b-4b71-8817-a56c822b13cd.png)

How can I color the field using U values, like this one:
![2](https://user-images.githubusercontent.com/57876661/156718787-a312223d-1630-4ad4-9725-e26464478937.png)

It seems that I can not upload the source vtk file due to the file type limit.

Sorry to ask such naive question. I'm really appreciated if anyone could help me to solve this problem! Thanks!

**Comment by whophil** [comment, marked as answer]
When you call `add_mesh()`, use `scalars='U Magnitude'`, or whatever the name of the field is.

```
plotter.add_mesh(mesh, scalars='U Magnitude')
```

**Reply by mellow753** [reply]
Thanks, it is exactly what I want! Another question is that my U=(u,v,w) vtk file is a vector (named 'U'). If I directly use `scalars='U'`, I am curious which component it plots, u or v or magnitude?

**Reply by tkoyama010** [reply]
By default, the magnitude is output. If you wish to output each component, use the component argument.
https://docs.pyvista.org/examples/02-plot/vector-component.html

**Reply by whophil** [reply]
I also do not know, but you can do this to just break it out into scalar fields.

```python
U = mesh['U']
mesh['u'] = mesh['U'][:, 0]
mesh['v'] = mesh['U'][:, 1]
mesh['w'] = mesh['U'][:, 2]
```

Maybe somebody else has a better solution.

**Reply by MatthewFlamm** [reply]
https://docs.pyvista.org/examples/02-plot/vector-component.html?highlight=component

Use:
```py
plotter.add_mesh(mesh, scalars='U', component=0)
```
